architecture/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,467 @@
+# Resource Extensibility (User Defined Resource Types)
+
+* **Author**: Reshma Abdul Rahim (@reshrahim)
+
+## Summary
+
+Radius is a core component of the internal developer platform for enterprises building cloud native applications. Many enterprises use a wide range of technologies together for achieving their cloud-native strategy. For any technology that’s a newcomer and helps solve a problem in the cloud native landscape, users look for an easy and seamless way to integrate their existing tools and technologies and incrementally adopt the new technology to their strategy. For Radius, we have heard requests from our users/community to support technologies that their applications are tightly coupled with E.g.: an internal messaging service or a technology they absolutely love E.g.: PostgreSQL/Kafka. Today, Radius provides [`Applications.Core/extenders`](https://docs.radapp.io/guides/author-apps/custom/overview/) to model any kind of service in an untyped way but they pose limitations for sophisticated enterprises in terms of guardrails and validation that platform engineers want to impose on their developer platform. 
```

Comment:
I would run this through a grammar check in Word. There is a lot little nits that Word would pickup real quick.